src/Stratis.Features.FederatedPeg/SourceChain/MaturedBlocksProvider.cs

```diff
@@ -66,6 +66,13 @@ public async Task<Result<List<MaturedBlockDepositsModel>>> GetMaturedDepositsAsy
 
                 ChainedHeaderBlock block = this.consensusManager.GetBlockData(currentHeader.HashBlock);
 
+                if (block?.Block?.Transactions == null)
+                {
+                    // Report unexpected results from consenus manager.
+                    this.logger.LogWarning("Stop matured blocks collection due to consensus manager integrity failure. Send what we've collected.");
```

Comment:
Can you explain this @quantumagi ?

Reply:
It's an unrelated issue discovered on one of the node. I just noticed apparent CM integrity issue  pop up once and thought I would add that message in case its related to the store stopping advancing sometimes. If it occurs again we will be able to make the connection.

Reply:
Note that we are planning to address this in the consensus rules too.